### ISSUE TITLE
net: re-enable or delete TestLookupLongTXT

### ISSUE THREAD

**Issue by 0intro** [golang/go#22857]
CL [79555](https://golang.org/cl/79555) added TestLookupLongTXT. This test is failing on Plan 9.

```
--- FAIL: TestLookupLongTXT (0.05s)
	lookup_test.go:323: LookupTXT golang.rsc.io incorrect
		have ["abcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXY"]
		want ["abcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXY" "gophers rule"]
FAIL
FAIL	net	23.605s
```

See https://build.golang.org/log/d83b9b31e94b0b5093c0d2813f78105531e8de3e

**Comment by 0intro** [comment]
It seems the Plan 9 DNS resolver (`ndb/dns`) only returns a single TXT record. It probably should be fixed.

```
cpu% ndb/dnsquery
> rsc.io ip
rsc.io ip	216.239.38.21
rsc.io ip	216.239.34.21
rsc.io ip	216.239.32.21
rsc.io ip	216.239.36.21
> golang.rsc.io txt
golang.rsc.io txt	gophers rule
> gandi.net txt
gandi.net txt	google-site-verification: OyQ-vUKOBlZ2onEvCpyG7UGmQC1_tahPOUSNqCWYruo
```

**Comment by bradfitz** [comment]
@0intro, feel free to send plan9 fixes whenever, but I'm removing this from Go1.10. If the fix only touches plan9 stuff, I'm comfortable taking it even late in the cycle.

**Comment by mvdan** [comment]
Different, potentially related failure: https://storage.googleapis.com/go-build-log/d38832a2/linux-amd64_b2d10b95.log

```
--- FAIL: TestLookupLongTXT (1.14s)
	lookup_test.go:342: LookupTXT golang.rsc.io incorrect
		have ["gophers rule"]
		want ["abcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXYabcdefghijklmnopqrstuvwxyABCDEFGHJIKLMNOPQRSTUVWXY" "gophers rule"]
```

This happened on an unrelated CL, https://go-review.googlesource.com/c/go/+/77253 - `linux-amd64 at d38832a2029f83e62a9ddfccfff6c2fc4e3a393e`.

**Comment by bradfitz** [comment]
This isn't just a plan9 issue. It's failing all over the place.

@ianlancetaylor and @rsc were discussing the problem recently. I think they understand what's happening here. Ian, can you summarize?

/cc @mikioh

**Comment by ianlancetaylor** [comment]
I don't actually understand what is going on here.  Maybe @rsc does.  It seems to have something to do with how caches handle multiple TXT records.

**Comment by ianlancetaylor** [comment]
This code was rewritten as part of #21160.  We should try enabling the test again at the start of the 1.13 cycle.

**Comment by ianlancetaylor** [comment]
See also #23873.

**Comment by bcmills** [comment]
> We should try enabling the test again at the start of the 1.13 cycle.

That didn't happen. Should it?

Marking as early-in-cycle release-blocker for Go 1.19.

**Comment by neild** [comment]
Looks like the same issue as described in https://github.com/golang/go/issues/22857#issuecomment-358027018 is still occurring:
https://build.golang.org/log/c7feb48ee4f491ec0c89462361e4b4fc763cd0ba
